### PR TITLE
[Docs] Update references in Ray Docs about batch inference

### PR DESCRIPTION
**Pull request by amogkam**
Updates various places batch inference is discussed in the Ray Docs to clarify what users should be looking at. Ray AIR batch prediction is out of scope for this PR.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/examples/batch_prediction.ipynb [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
     "* ML model: this is a trained ML model which is usually also stored in an external storage system.\n",
     "* Predictions: these are the outputs when applying the ML model on observations. The predictions are usually written back to the storage system.\n",
     "\n",
-    "With Ray, you can build scalable batch prediction for large datasets at high prediction throughput. This is achieved by splitting the dataset into disjoint shards and executing them in parallel, with either Ray tasks or Ray actors across a Ray cluster.\n",
-    "```\n",
-    "\n",
+    "With Ray, you can build scalable batch prediction for large datasets at high prediction throughput. Ray Data provides a [higher-level API for offline batch inference](batch_inference_home), with many optimizations built-in. However, for more control, you can use the lower-level Ray Core APIs. In this example we do batch inference with Ray Core by splitting the dataset into disjoint shards and executing them in parallel, with either Ray tasks or Ray actors across a Ray cluster."
```
Comment (angelinalg) [inline doc/source/ray-core/examples/batch_prediction.ipynb:19]:
```suggestion
    "With Ray, you can build scalable batch prediction for large datasets at high prediction throughput. Ray Data provides a [higher-level API for offline batch inference](batch_inference_home), with many optimizations built-in. However, for more control, you can use the lower-level Ray Core APIs. This example demonstrates batch inference with Ray Core by splitting the dataset into disjoint shards and executing them in parallel, with either Ray Tasks or Ray Actors across a Ray Cluster."
```

File: doc/source/serve/tutorials/batch.md [hunk 2]
```diff
@@ -9,14 +9,9 @@ a batch of queries and processes them at once. In particular, we show:
...
 
-This tutorial should help the following use cases:
-
-- You want to perform offline batch inference on a cluster of machines.
-- You want to serve online queries and your model can take advantage of batching.
-  For example, linear regressions and neural networks use CPU and GPU's
-  vectorized instructions to perform computation in parallel. Performing
-  inference with batching can increase the *throughput* of the model as well as
-  *utilization* of the hardware.
+This tutorial should help when you want to serve online queries and your model can take advantage of batching. For example, linear regressions and neural networks use CPU and GPU's vectorized instructions to perform computation in parallel. Performing inference with batching can increase the *throughput* of the model as well as *utilization* of the hardware.
```
Comment (angelinalg) [inline doc/source/serve/tutorials/batch.md:12]:
```suggestion
This tutorial is a guide for serving online queries when your model can take advantage of batching. For example, linear regressions and neural networks use CPU and GPU's vectorized instructions to perform computation in parallel. Performing inference with batching can increase the *throughput* of the model as well as *utilization* of the hardware.
```

File: doc/source/serve/tutorials/batch.md [hunk 3]
```diff
@@ -9,14 +9,9 @@ a batch of queries and processes them at once. In particular, we show:
...
+
+For _offline_ batch inference with large datasets, see [batch inference with Ray Data](batch_inference_home) instead.
```
Comment (angelinalg) [inline doc/source/serve/tutorials/batch.md:14]:
```suggestion
For _offline_ batch inference with large datasets, see [batch inference with Ray Data](batch_inference_home).
```

**Review by angelinalg — APPROVED** [review]

**Review by shrekris-anyscale — APPROVED** [review]
Nice work, the Serve changes look good to me!

**Pull request merged**